akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -329,29 +334,27 @@ import scala.util.control.NonFatal
       require(maxBuffer > 0, "Buffer size must be greater than 0")
 
       var buffer: Buffer[Received[T]] = _
-      var currentRequest: Option[Requested[T]] = None
+      var currentRequests: Buffer[Requested[T]] = _
 
       override def preStart(): Unit = {
         // Allocates one additional element to hold stream
         // closed/failure indicators
         buffer = Buffer(maxBuffer + 1, inheritedAttributes)
+        currentRequests = Buffer(maxConcurrentPulls, inheritedAttributes)
```

Comment:
I think this is a left over of some old problem with buffers, please move both buffer creations to the constructor and make the fields `val` instead of var.

Reply:
Fixed